master/internal/templates/api.go

```diff
@@ -0,0 +1,259 @@
+package templates
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"strings"
+
+	"github.com/pkg/errors"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/determined-ai/determined/master/internal/api"
+	"github.com/determined-ai/determined/master/internal/authz"
+	"github.com/determined-ai/determined/master/internal/db"
+	"github.com/determined-ai/determined/master/internal/grpcutil"
+	"github.com/determined-ai/determined/master/pkg/model"
+	"github.com/determined-ai/determined/proto/pkg/apiv1"
+	"github.com/determined-ai/determined/proto/pkg/templatev1"
+)
+
+// TemplateAPIServer implements the template APIs for Determined's API server.
+type TemplateAPIServer struct{}
+
+// GetTemplates viewable by the user. If there are no matches, returns an empty list.
+func (a *TemplateAPIServer) GetTemplates(
+	ctx context.Context,
+	req *apiv1.GetTemplatesRequest,
+) (*apiv1.GetTemplatesResponse, error) {
+	user, _, err := grpcutil.GetUser(ctx)
+	if err != nil {
+		return nil, status.Errorf(codes.Unauthenticated, "failed to get the user: %s", err)
+	}
+
+	scopes, err := AuthZProvider.Get().ViewableScopes(ctx, user, 0)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "failed to check for permissions: %s", err)
+	}
+
+	var resp apiv1.GetTemplatesResponse
+	if err := db.Bun().NewRaw(`
+SELECT name, config, workspace_id
+FROM templates
+`).Scan(ctx, &resp.Templates); err != nil {
```

Comment:
nit: could replace with `err: db.Bun().NewSelect().Table("templates").Column("name", "config", "workspace_id").Scan(ctx, &resp.Templates)`

Reply:
went with `ColumnExpr("*")` so if someone adds a column the test will break if they don't add it to the response

Reply:
went with the bun query builder style so I can push the pagination/sorting/etc to the DB easier.